x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/lib/get_time_options.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { getTimeUnitLabel } from './get_time_unit_label';
+import { TIME_UNITS } from '../constants';
+
+export const getTimeOptions = (unitSize: string) =>
+  Object.entries(TIME_UNITS).map(([_key, value]) => {
+    return {
+      text: getTimeUnitLabel(value, unitSize),
+      value,
+    };
+  });
+
+export const getTimeFieldOptions = (fields: any, firstFieldOption: any) => {
+  const options = [firstFieldOption];
```

Comment:
It feels weird that we're providing `firstFieldOption` from outside and all we do is push it into the array first.
If the code calling this function knows what shoulkd come first the nperhaps it should push this int othe array after this function returns? Providing it as an argument means you can't use `getTimeFieldOptions` unless you _have_ something you want to put in first (it's a required argument) so we're actually limiting how this function can be used, while also (IMHO) leaking implementation logic around.